src/pages/sidepanel/index.tsx

```diff
@@ -8,18 +10,64 @@ function init() {
   if (!appContainer) {
     throw new Error('Can not find #app-container');
   }
+
+  ReactDOM.render(<IndexSidepanel />, appContainer);
 }
 
 init();
 
-chrome.runtime.sendMessage({ greeting: 'Hello' }, function (response) {
-  console.log(response.farewell);
-});
-
-chrome.runtime.onMessage.addListener((message, _sender, sendResponse) => {
-  if (message.action === 'getContent') {
-      const content = 'This is the content from the background script';
-      sendResponse({content: content});
+chrome.runtime.onMessage.addListener(
+  function(request, sender, sendResponse) {
+    console.log(sender.tab ? "from a content script:" + sender.tab.url : "from the extension");
+    if (request.greeting === "hello") sendResponse({farewell: "goodbye"});
   }
-  return true;
-});
+);
+
+function IndexSidepanel(): JSX.Element {
+  const [currentUrl, setCurrentUrl] = useState<string>("");
+  const [formData, setFormData] = useState<{ name: string, email: string }>({ name: '', email: '' });
+
+  const getCurrentUrl = async (): Promise<void> => {
+    const [tab] = await chrome.tabs.query({ active: true, currentWindow: true });
+    setCurrentUrl(tab.url || 'unknown');
+  };
+
+  useEffect(() => {
+    getCurrentUrl();
+  }, [currentUrl]);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormData(prevState => ({ ...prevState, [name]: value }));
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
+      if (tabs[0]?.id) {
+        chrome.tabs.sendMessage(tabs[0].id, { action: "fillForm", data: formData });
+      }
+    });
+  };
+
+  return (
+    <div>
```

Comment:
This HTML should only show all attributes of the form fields like. name, id,  type (text, email, number), value.